File: 0501_0600/0534/0534.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    p = [int(x) for x in f.readline().split()]
    f.readline()
    s = f.readline().split()
    d = [s.count(str(i)) for i in range(1, n + 1)]
    q.write("\n".join("yes" if d[i] > p[i] else "no" for i in range(n)))

File: 0501_0600/0534/0534.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, k, d, *p;

	fscanf(f, "%d", &n);
	p = (int*)malloc(sizeof(int) * n);
	for (int i = 0; i < n; ++i)
		fscanf(f, "%d", p + i);
	fscanf(f, "%d", &k);
	while (k--)
	{
		fscanf(f, "%d", &d);
		--p[d - 1];
	}
	for (int i = 0; i < n; ++i)
		fprintf(q, p[i] < 0 ? "yes\n" : "no\n");
	return 0;
}
